packages/mgt-components/src/components/mgt-todo/mgt-todo.ts

```diff
@@ -566,24 +639,78 @@ export class MgtTodo extends MgtTasksBase {
     }
   }
 
+  private handleTaskCheckKeydown(e: KeyboardEvent, task: TodoTask) {
+    if (e.key === 'Enter' && !this.readOnly) {
+      this.handleTaskClick(task);
+      if (task.status === 'completed') {
+        void this.updateTaskStatus(task, 'notStarted');
+      } else {
+        void this.updateTaskStatus(task, 'completed');
+      }
+    }
+  }
+
   private readonly handleInput = (e: MouseEvent) => {
     if ((e.target as HTMLInputElement).id === 'new-task-name-input') {
       this._newTaskName = (e.target as HTMLInputElement).value;
     }
   };
 
+  private readonly handleChange = async (e: KeyboardEvent, task: TodoTask) => {
+    if ((e.target as HTMLInputElement).id === task.id) {
+      if (e.key === 'Enter') {
+        await this.updateTask(task);
+        (e.target as HTMLInputElement)?.blur();
+      }
+      this._changedTaskName = (e.target as HTMLInputElement).value;
+    }
+  };
+
   private readonly handleKeyDown = async (e: KeyboardEvent) => {
-    if (e.key === 'Enter') {
+    if (e.key === 'Enter' && (e.target as HTMLInputElement).id === 'new-task-name-input') {
       await this.addTask();
     }
   };
 
+  private readonly updatingTask = (e: KeyboardEvent, task: TodoTask) => {
+    if ((e.target as HTMLInputElement).id === task.id) {
+      this._taskBeingUpdated = task;
+    }
+    if ((e.target as HTMLInputElement).id === `${task.id}-taskDate-input`) {
+      this._updatingTaskDate = true;
+      this._taskBeingUpdated = task;
+    }
+  };
+
+  private readonly handleBlur = () => {
+    const task = this._taskBeingUpdated;
+    const targets = this.renderRoot.querySelectorAll('fluent-text-field');
+    for (const target of targets) {
+      if (
+        task &&
+        ((target as HTMLInputElement).id === task.id || (target as HTMLInputElement).id === `${task.id}-taskDate-input`)
+      ) {
+        void this.updateTask(task);
+        (target as HTMLInputElement)?.blur();
+        this._taskBeingUpdated = null;
+        this._updatingTaskDate = false;
+      }
+    }
+  };
+
   private readonly handleDateChange = (e: Event) => {
     const value = (e.target as HTMLInputElement).value;
     if (value) {
       this._newTaskDueDate = new Date(value + 'T17:00');
     } else {
       this._newTaskDueDate = null;
     }
+
+    const oldValue = new Date(this._taskBeingUpdated.dueDateTime?.dateTime).toLocaleDateString();
```

Comment:
The order of operations here suggests that there could be a case where we try and construct an oldValue date based on a nullish `this._taskBeinUpdated` value.

suggest that we re-work this whole method so that the opening of the if statement is the first line of the method and all the logic is gated on actually having a _taskBeingUpdated value

Reply:
I'm re-using this method when adding a new task and updating a new task. In the case of adding a new task, there is no _taskBeingUpdated value. I could create different methods for the different cases.

Reply:
Oh, that makes sense, sorry I didn't catch that this was to track the value changes for _newTaskDueDate as well, in that case only the assignment to oldValue would be inside the if block